### ISSUE TITLE
Fields.get(Name) should allow for accessing partially qualified names

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13939]
This idea has last been seen here: https://github.com/jOOQ/jOOQ/issues/13849

There's a historic feature where you could name your columns like this:

```java
List<P> result =
ctx.select(BOOK.ID.as("x.y.id"))
   .from(BOOK)
   .fetchInto(P.class);
```

And define:

```java
record Y (int id) {}
record X (Y y) {}
record P (X x) {}
```

Now by convention, the mapping as implemented by `DefaultRecordMapper` would work as expected.

Some users' expectation here is to be able to refer to the column in the record by `"x.y.id"`, `"y.id"`, or `"id"`, as if the column had a qualified name. E.g. if we're fetching things into `Record` instead of into `P`, we could then write:

```java
Record r = ...

// These would all be the same:
r.get("x.y.id");
r.get("y.id");
r.get("id");
```

All of these names are still unqualified `org.jooq.Name` expressions, except that they contain the `Settings.namePathSeparator` character, which has been used by jOOQ for various nesting purposes, as the above `DefaultRecordMapper` nesting.

It's not yet clear what the implications of such a change will be. If implemented, we'll have to take into account:

- [ ] How this implicit nesting plays with explicit nesting of records
- [ ] How we can access `Settings.namePathSeparator` in case the record is detached
- [ ] How this implicit nesting plays with the qualification of the field in a table/schema/etc and how to resolve ambiguities if there is also a schema `x`, table `y` and column `id` in the record
- [ ] Others?

At this point, it's not yet clear if we should really support this.

**Comment by alf** [comment]
Stumbled upon this as I was trying to implement some subtyping-functionality.

We want one select query against the database that returns multiple subtypes and thus need to map the resulting data to the concrete subtypes in Java. In this example I'm only looking at the "all subtypes in one table"-pattern.

Here's what I tried. The details here are not really important, the gist is that I want to use the first column of the result record for the pagination token and the second column for the actual data. Since I expect that I have to support more than 22 columns for my actual data I prefer to work against the Record interface.

```java
return ctx
                .select(getOrderByToken(API_HENDELSE, API_HENDELSE.fields(API_HENDELSE.ID)),
                        row(
                            API_HENDELSE.INSTITUSJONSNR_EIER,
                            API_HENDELSE.ID,
                            API_HENDELSE.TIDSPUNKT,
                            API_HENDELSE.HENDELSESTYPEKODE,
                            API_HENDELSE.PK,
                            API_HENDELSE.HENDELSESDATA
                        ).convertFrom(StudenthendelseService::intoStudenthendelseSubtype)
                )
                .from(API_HENDELSE)
                .where(API_HENDELSE.HENDELSESTYPEKODE.in(hendelsestyper))
                .and(API_HENDELSE.INSTITUSJONSNR_EIER.eq(eierInstitusjonsnummer))
                .and(studnivinetervallCondition(API_HENDELSE, studienivabetegnelser))
                .orderBy(API_HENDELSE.fields(API_HENDELSE.ID))
                .seek(getOrderByValues(ctx, API_HENDELSE.fields(API_HENDELSE.ID), token))
                .limit(pageSize)
                .fetch();
```

Unfortunately I get in trouble since I can no longer access the data in my row-expression the way I would like:

```java
private static Studenthendelse intoStudenthendelseSubtype(Record6<String, BigInteger, OffsetDateTime, String, JSON, JSON> rec) {
        switch (rec.get(API_HENDELSE.HENDELSESTYPEKODE)) {
            case "ENDRET_FOLKEREGISTRERT_ADRESSE":
                return rec.into(StudentEndretFolkeregistrertAdresse.class);
            case "ENDRET_NAVN":
                return rec.into(StudentEndretNavn.class);
    ...
```

While this isn't a blocker, I figured you'd be interested in knowing that I stumbled here.

**Comment by lukaseder** [comment]
Thanks for your comment, @alf. I don't think this is the same issue. The issue here is specifically about using the path-in-name convention in order to access a `DefaultRecordMapper` feature, where the mapping can nest data based on this naming convention.

You're thinking about this, instead?

- https://github.com/jOOQ/jOOQ/issues/16721

**Comment by alf** [comment]
Indeed. That’s a better issue for me to follow. Thank you, I’ll follow up with a question there. :)
